azurelinuxagent/common/cgroups.py

```diff
@@ -267,6 +267,26 @@ def stop_tracking(name):
         if CGroupsTelemetry.is_tracked(name):
             del (CGroupsTelemetry._tracked[name])
 
+    @staticmethod
+    def get_cpu_limits(name):
+        # default values
+        cpu_limit = DEFAULT_CPU_LIMIT_AGENT if AGENT_NAME.lower() in name.lower() else DEFAULT_CPU_LIMIT_EXT
+
+        return cpu_limit
+
+    @staticmethod
+    def get_memory_limits(name):
+        osutil = get_osutil()
+
+        # default values
+        mem_limit = max(DEFAULT_MEM_LIMIT_MIN_MB, round(osutil.get_total_mem() * DEFAULT_MEM_LIMIT_PCT / 100, 0))
+
+        # agent values
+        if AGENT_NAME.lower() in name.lower():
+            mem_limit = min(DEFAULT_MEM_LIMIT_MAX_MB, mem_limit)
+
+        return mem_limit
+
```

Comment:
DRY code smell. You now compute these limits _twice_; once when you collect results, and once when you actually supply the limits to the OS cgroups implementation.

This is not the right way to do it.

Reply:
Yup - I didn't want to do it. Please go over my comment [below](https://github.com/Azure/WALinuxAgent/pull/1394#discussion_r232926231) and I can make this better.